### ISSUE TITLE
Issues when using Next.js with basePath

### ISSUE THREAD

**Issue by renet** [nibtime/next-safe-middleware#48]
When using Next.js (18) with a basePath set in next.config.js (in our case it has a value of "/dashboard"), it seems like the `script-hashes.txt` and `style-hashes.txt` files cannot be accessed by the middleware when processing the requests. This results in errors like

```
TypeError: default-src 'self';object-src 'none';base-uri 'none';connect-src 'self' ws://localhost:3000/;font-src 'self';img-src 'self' data:;script-src 'unsafe-inline' <!DOCTYPE html><html><!-- full content of the 404 page --></html> 'unsafe-hashes' is not a legal HTTP header value
```

Using Next.js > 12.2, yarn 3.2.1 and Node 16. If you need any further information, please let me know. And thanks for a great project! :)

**Comment by nibtime** [comment]
Hi @renet 

thanks for reporting this! Sorry for the little late reply, I was busy with a PR (#47) to fix stuff for Next 12.2 and other things lately.  Fortunately, this one was an easy fix, it will be included in the `0.9.0` release. Also recommended for lots of other 12.2 fixes and when you host on Vercel because of #45.

**Comment by nibtime** [comment]
@all-contributors add @renet for bug
